src/storage/user-data.ts

```diff
@@ -46,6 +46,19 @@ function populatePhpVersion( sites: SiteDetails[] ) {
 	} );
 }
 
+function populateSnapshotUserIds( data: UserData ): void {
+	const userId = data.authToken?.id;
+
+	if ( userId && data.snapshots ) {
+		data.snapshots = data.snapshots.map( ( snapshot ) => {
+			if ( ! snapshot.userId ) {
+				return { ...snapshot, userId };
+			}
+			return snapshot;
+		} );
+	}
+}
```

Comment:
We are adding this function that assigns a userId to pre-existing snapshots so they appear to in the previews tab.
We can consider removing this code after a few weeks/months as those preview sites would expire.
Before this PR, if a user logged out the snapshots disappeared from Studio.

Reply:
Wojtek suggested that we could rename the function and add a little bit of context instead of leaving the task hanging, so I updated the task and created related PR: https://github.com/Automattic/studio/pull/954.